easybuild/framework/easyconfig/format/one.py

```diff
@@ -312,64 +340,134 @@ def extract_comments(self, rawtxt):
         Inline comments on items of iterable values are also extracted.
         """
         self.comments = {
-            'above': {},  # comments for a particular parameter definition
+            'above': {},  # comments above a parameter definition
             'header': [],  # header comment lines
             'inline': {},  # inline comments
-            'iter': {},  # (inline) comments on elements of iterable values
-            'tail': [],
+            'iterabove': {},  # comment above elements of iterable values
+            'iterinline': {},  # inline comments on elements of iterable values
+            'tail': [],  # comment at the end of the easyconfig file
          }
 
-        rawlines = rawtxt.split('\n')
+        parsed_ec = self.get_config_dict()
 
-        # extract header first
-        while rawlines and rawlines[0].startswith('#'):
-            self.comments['header'].append(rawlines.pop(0))
+        param_def_regex = re.compile(r'^([a-z__0-9]+)\s*=')
```

Comment:
Why the double "_" in the regex?

Reply:
Ah, shouldn't be there. Doesn't matter much, but I'll clean it up.